src/test/java/com/nimbusds/jose/JWEObjectJSONTest.java

```diff
@@ -0,0 +1,354 @@
+package com.nimbusds.jose;
+
```

Comment:
It would be great to add the following tests
1) Custom per-recipient header is passed (with kid specified) for encryption
2) kid value is passed (in addition to the key) to decryption
3) Negative test decryption: keys from different curves (as in non-json case)
4) Negative test decryption: ciphertext is invalid (exception is expected)
5) Negative test decryption: passed recipient kid is not found in the per-recipient headers
6) Negative test decryption: no epk in decrypt header
7) Negative test decryption: encrypted_key value for the the given recipient key is invalid (exception expected)
8) Negative test decryption: encrypted_key value for another recipient (not passed) is invalid (no exception expected?)
9) Negative test decryption: unexpected values in JSON are met (aad in particular)
10) Test decryption: multi-recipient encryption for two keys is done, but decryption is done for one key only (separately for Bob and for Charlie)
11) Negative test decryption: decryption for two keys, where the first key can be unwrapped, but the second can not (invalid encrypted_key value), exception is expected?